utils/sysfs/sysfs.go

```diff
@@ -335,25 +348,198 @@ func (fs *realSysFs) GetSystemUUID() (string, error) {
 	}
 }
 
-func (fs *realSysFs) IsCPUOnline(dir string) bool {
-	cpuPath := fmt.Sprintf("%s/online", dir)
-	content, err := ioutil.ReadFile(cpuPath)
+func (fs *realSysFs) IsCPUOnline(cpuPath string) bool {
+	return isCPUOnline(cpuPath)
+}
+
+// isCPUOnline verifies if a CPU in online. Takes path to /sys/devices/system/cpu/cpu*/ directory as a parameter.
+func isCPUOnline(cpuPath string) bool {
+	if isX86 && isZeroCPU(cpuPath) {
```

Comment:
Can we add a comment that describes why the `isX86` check is needed?

As I understand the problem is that assumption that zero cpu is always online isn't true for some non x86 systems, e.g. from the rpi4 you collected:

```
$ cat utils/sysfs/testdata_rpi4/cpu0/online
cat: utils/sysfs/testdata_rpi4/cpu0/online: No such file or directory
```

side point: what is the reason that we need to special case cpu0 again? For example, even if cpu0 online file doesn't exist (like on rpi4), would you be able to rely on the backup hotplug check? It seems like it's there, for example:

```
$ cat utils/sysfs/testdata_rpi4/hotplug/states  | grep -E "(\d)*: online$"
206: online
$ cat utils/sysfs/testdata_rpi4/cpu0/hotplug/state
206
```